cirq/google/engine/engine_processor.py

```diff
@@ -167,6 +170,132 @@ def get_current_calibration(self,) -> Optional[calibration.Calibration]:
         else:
             return None
 
+    def create_reservation(self, start_time: datetime.datetime,
+                           end_time: datetime.datetime,
+                           whitelisted_users: List[str]):
+        """Creates a reservation on this processor.
+
+        Args:
+            start_time: the starting date/time of the reservation.
+            end_time: the ending date/time of the reservation.
+            whitelisted_users: a list of emails that are allowed
+              to send programs during this reservation (in addition to users
+              with permission "quantum.reservations.use" on the project).
+        """
+        response = self.context.client.create_reservation(
+            self.project_id, self.processor_id, start_time, end_time,
+            whitelisted_users)
+        return response
+
+    def _delete_reservation(self, reservation_id: str):
+        """Delete a reservation.
+
+        This will only work for reservations outside the processor's
+        schedule freeze window.  If you are not sure whether the reservation
+        falls within this window, use remove_reservation
+        """
+        return self.context.client.delete_reservation(self.project_id,
+                                                      self.processor_id,
+                                                      reservation_id)
+
+    def _cancel_reservation(self, reservation_id: str):
+        """Cancel a reservation.
+
+        This will only work for reservations inside the processor's
+        schedule freeze window.  If you are not sure whether the reservation
+        falls within this window, use remove_reservation
+        """
+        return self.context.client.cancel_reservation(self.project_id,
+                                                      self.processor_id,
+                                                      reservation_id)
+
+    def remove_reservation(self, reservation_id: str):
+        reservation = self.get_reservation(reservation_id)
+        if reservation is None:
+            raise ValueError(f'Reservation id {reservation_id} not found.')
+        if self._processor:
```

Comment:
It looks like _self._inner_procecessor() should lazy load the _processor if it hasn't been fetched already.

Reply:
Fixed.